pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -185,20 +186,16 @@ def _run_optimization(
         else:
             if self.trainer.lightning_module.automatic_optimization:
                 self._optimizer_step(optimizer, opt_idx, batch_idx, closure)
-                if len(self.trainer.optimizers) > 1:
-                    # revert back to previous state
-                    self.trainer.lightning_module.untoggle_optimizer(opt_idx)
             else:
                 result = self._training_step(split_batch, batch_idx, opt_idx, self._hiddens)
 
-            if not result:
-                # user decided to skip optimization
-                return result
```

Comment:
reviewer: refactored this so we have a single exit point out of the method, and we can consistently run untoggle_optimizer before exiting to ensure we always have toggles paired with untoggles.